partiql-planner/src/test/kotlin/org/partiql/planner/internal/transforms/OrderByAliasSupportTest.kt

```diff
@@ -0,0 +1,465 @@
+                    UNION
+                    SELECT col AS alias FROM t2
+                    ORDER BY alias
+                """,
+                description = "UNION with ORDER BY alias resolution"
+            ),
+            SuccessTestCase(
+                name = "union_with_order_by_alias_with_inner_order_by",
+                sql = """
+                    (SELECT a AS c FROM tA ORDER BY c LIMIT 4 OFFSET 1)
+                    UNION
+                    (SELECT b AS c FROM tB ORDER BY c LIMIT 4 OFFSET 3)
+                    ORDER BY c DESC LIMIT 100 OFFSET 1
+                """,
+                transformedSql = """
+                    (SELECT a AS c FROM tA ORDER BY a LIMIT 4 OFFSET 1)
+                    UNION
+                    (SELECT b AS c FROM tB ORDER BY b LIMIT 4 OFFSET 3)
+                    ORDER BY c DESC LIMIT 100 OFFSET 1
+                """,
+                description = "UNION with ORDER BY alias resolution"
+            ),
+            SuccessTestCase(
+                name = "intersect_with_order_by_alias",
+                sql = """
+                    SELECT price AS p FROM products
+                    INTERSECT
+                    SELECT cost AS p FROM orders
+                    ORDER BY p
+                """,
+                transformedSql = """
+                    SELECT price AS p FROM products
+                    INTERSECT
+                    SELECT cost AS p FROM orders
+                    ORDER BY p
+                """,
+                description = "INTERSECT with ORDER BY alias resolution"
+            ),
+            SuccessTestCase(
+                name = "except_with_order_by_alias",
+                sql = """
+                    SELECT id AS identifier FROM table1
+                    EXCEPT
+                    SELECT id AS identifier FROM table2
+                    ORDER BY identifier
+                """,
+                transformedSql = """
+                    SELECT id AS identifier FROM table1
+                    EXCEPT
+                    SELECT id AS identifier FROM table2
+                    ORDER BY identifier
+                """,
+                description = "EXCEPT with ORDER BY alias resolution"
+            ),
+            SuccessTestCase(
+                name = "union_all_with_complex_alias",
+                sql = """
+                    SELECT a + b AS total FROM t1
+                    UNION ALL
+                    SELECT x + y AS total FROM t2
+                    ORDER BY total DESC
+                """,
+                transformedSql = """
+                    SELECT a + b AS total FROM t1
+                    UNION ALL
+                    SELECT x + y AS total FROM t2
+                    ORDER BY total DESC
+                """,
+                description = "UNION ALL with complex expression alias resolution"
+            )
+        )
+
+        @JvmStatic
+        fun failedTestCases() = listOf(
+            FailedTestCase(
```

Comment:
Let's add both cases that were mentioned in the other comment https://github.com/partiql/partiql-lang-kotlin/pull/1831#discussion_r2402971346 -- case-insensitive ambiguity and case-sensitive ambiguity.